compiler/ballerina-parser/src/test/java/io/ballerinalang/compiler/parser/test/syntax/expressions/QueryExpressionTest.java

```diff
@@ -169,4 +169,16 @@ public void testQueryWithOperatorPrecedenceWithExtraClauses() {
         test("from int a in b select c let int a = b + from int f in g select h",
                 "query-expr/query_expr_assert_38.json");
     }
+
+    @Test
+    public void testQueryWithOrderByClause() {
+        test("from int a in b where c let int d = e order by f select g", "query-expr/query_expr_assert_39.json");
+        test("from int a in b where c let int d = e order by f ascending, g select h", "query-expr" +
+                "/query_expr_assert_40.json");
+        test("from int a in b order by c ascending, d descending select e", "query-expr" +
+                "/query_expr_assert_41.json");
+        test("from int a in b order by select c", "query-expr/query_expr_assert_42.json");
+        test("from int a in b order c select d", "query-expr/query_expr_assert_43.json");
+        test("from int a in b order by c ascending d select e", "query-expr/query_expr_assert_44.json");
+    }
```

Comment:
does this includes negative tests (tests with incorrect syntaxes) also? 
 - If yes - shall we split it in to two? There are separate sections in the same test file for positive tests and negative tests
- If not - shall we add some?